### PR TITLE
Don't add a magic trailing comma for a single entry

### PR DESCRIPTION
**Pull request by konstin**
## Summary

If a comma separated list has only one entry, black will respect the magic trailing comma, but it will not add a new one.  

The following code will remain as is:

```python
b1 = [
    aksjdhflsakhdflkjsadlfajkslhfdkjsaldajlahflashdfljahlfksajlhfajfjfsaahflakjslhdfkjalhdskjfa
]
b2 = [
    aksjdhflsakhdflkjsadlfajkslhfdkjsaldajlahflashdfljahlfksajlhfajfjfsaahflakjslhdfkjalhdskjfa,
]
b3 = [
    aksjdhflsakhdflkjsadlfajkslhfdkjsaldajlahflashdfljahlfksajlhfajfjfsaahflakjslhdfkjalhdskjfa,
    aksjdhflsakhdflkjsadlfajkslhfdkjsaldajlahflashdfljahlfksajlhfajfjfsaahflakjslhdfkjalhdskjfa
]
```

## Test Plan

This was first discovered in https://github.com/django/django/blob/7eeadc82c2f7d7a778e3bb43c34d642e6275dacf/django/contrib/admin/checks.py#L674-L681, which i've minimized into a call test. 

I've added tests for the three cases (one entry + no comma, one entry + comma, more than one entry) to the list tests.

The diffs from the black tests get smaller.

### CODE REVIEW

**Comment by konstin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5463** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5463" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5463?utm_source=stack-comment).

File: crates/ruff_python_formatter/src/builders.rs [hunk 1]
```diff
@@ -183,6 +183,9 @@ pub(crate) struct JoinCommaSeparatedBuilder<'fmt, 'ast, 'buf> {
     result: FormatResult<()>,
     fmt: &'fmt mut PyFormatter<'ast, 'buf>,
     last_end: Option<TextSize>,
+    /// We need to track whether we have more than one entry since a sole entry doesn't get a
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/builders.rs:186]: Nit: Tracking the whole length and last_end results in redundant information. How about an enum with the variants First, Second(TextSize), Rest(TextSize)?

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_python_formatter/src/builders.rs (hunk 1 above)
Comment (konstin) [reply crates/ruff_python_formatter/src/builders.rs:186]: I could offer 
```rust

struct LastEntry {
    end: TextSize,
    only_entry: bool
}

last_entry: Option<LastEntry>
```

but i'm not sure whether that makes it better. (the enum makes the code hard to understand)

Comment (konstin) [reply crates/ruff_python_formatter/src/builders.rs:186]: i'll merge this now because it blocks import formatting, tell me if you still want this

**Pull request merged**